driver/src/price_finding/stablex_naive_solver.rs

```diff
@@ -0,0 +1,242 @@
+use crate::price_finding::error::PriceFindingError;
+use crate::price_finding::PriceFinding;
+use crate::util;
+use dfusion_core::models::{AccountState, Order, Solution};
+use std::cmp::Ordering;
+
+/// A naive solver for the stableX contract. It works almost identically to the
+/// Snapp naive solver except it has a few extra requirements and it doesn't try
+/// to be as optimal.
+///
+/// The solver works by finding the first pair of orders which matching sell and
+/// buy tokens, where one of the traded tokens is the fee token, whose users have
+/// enough balance to fully fill, and that has a viable solution.
+///
+/// The solution is computed by taking the price of the larger order and and
+/// completely filling the smaller order. A large order is defined by having
+/// more non-fee token volume. Account balances are checked for each solution.
+pub struct StableXNaiveSolver;
```

Comment:
Why has this been merged, if https://github.com/gnosis/dex-services/pull/297#pullrequestreview-318346510 is still open?

Reply:
We ended getting into an issue that we needed the new naive solver to merge the objective solution PR to make the e2e tests pass, but the naive solver needed updates to `dex-contracts` to include `close-auction` in order to reliably pass e2e tests (I believe master was failing, it was caused by the timing changes to the contracts that got merged in). And the `dex-contract` updates were done on latest so they needed the services to be updated to the new contract interface (which is what the objective value PR was).

Essentially we had a circular dependency of PRs. What we could have done would be to apply the `close_acution` script changes to the commit that master was based on, merge the naive solver, then merge the objective value PR. However, we both the naive solver PRs had been reviewed independently in #297 and #296 so we decided to just join them so that we could get things building again.